### DISCUSSION TITLE
npm package to deno converter

### DISCUSSION THREAD

**Discussion by phocodes** [denoland/deno#13653] [Q&A]
A way to turn existing npm packages so that they can be compatible with deno.

**Comment by kitsonk** [comment]
If you are just wanting to use an existing npm package under Deno, the easiest way currently is to use a CDN to import from, like esm.sh or skypack.dev or esm.run.  While experimental, many npm packages can be run under Deno with the `--compat` mode flag without modification as well.

If you are looking to convert your existing npm package to Deno, including all the CI and scripts, there isn't anything specifically provided and supported by the core team, though we are continuing to explore this space.

To take a Deno "package" and export it to npm for consumption in Node.js, then there is [dnt](https://github.com/denoland/dnt).

**Reply by sunw31** [reply]
i have a idea, deno 1.x compatible to npm, deno 2.x can not compatible to npm.

npm is a large ecosystem, includes web3 infra package...

we love deno
